pkg/sql/pgwire/command_result.go

```diff
@@ -327,11 +328,16 @@ func (c *conn) newCommandResult(
 		return r
 	}
 	telemetry.Inc(sqltelemetry.PortalWithLimitRequestCounter)
+	suspensionStrategy := nonSelectPortalSuspensionStrategy
+	if stmtTag == "SELECT" {
```

Comment:
I don't think this is correct. See https://github.com/postgres/postgres/blob/2f48ede080f42b97b594fb14102c82ca1001b80c/src/backend/tcop/pquery.c#L227. There it looks at two things to determine the ONE_SELECT portal strategy: it must have a commandType of CMD_SELECT and not be have a modifying CTE (https://github.com/postgres/postgres/blob/2f48ede080f42b97b594fb14102c82ca1001b80c/src/backend/parser/parse_cte.c#L147). Here we are only checking for the tag, not the commandType.